Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -102,8 +102,9 @@ def cmd(*args)
   def cmd_fail(*args)
     output = cmd_output(*args)
     status = $?.exitstatus
-    $stderr.puts "\n#{output}" if status != 1
-    assert_equal 1, status
+    $stderr.puts "\n#{output}" if status < 1 || status > 3
+    assert_operator status, :>, 0
+    assert_operator status, :<, 4
```

Comment:
The implicit assumption that is always made on the command line is that any nonzero exit code counts as failure. For this reason, the best way to address the too restrictive assumption previously made here would be to simply consider every nonzero exit code a failure (in the sense that the command failed) and thus a success (in terms of that the failure was expected in the test).

Or to be a bit more specific, this should do (and yes, ideally is done in a separate commit):

```ruby
$stderr.puts "\n#{output}" if status == 0
refute_equal 0, status
```